theatre/studio/src/StudioStore/createTransactionPrivateApi.ts

```diff
@@ -0,0 +1,248 @@
+import type {Pointer} from '@theatre/dataverse'
+import {isSheetObject} from '@theatre/shared/instanceTypes'
+import type {$FixMe, $IntentionalAny} from '@theatre/shared/utils/types'
+import get from 'lodash-es/get'
+import type {ITransactionPrivateApi} from './StudioStore'
+import forEachDeep from '@theatre/shared/utils/forEachDeep'
+import getDeep from '@theatre/shared/utils/getDeep'
+import type {SequenceTrackId} from '@theatre/shared/utils/ids'
+import {getPointerParts} from '@theatre/dataverse'
+import type {
+  PropTypeConfig,
+  PropTypeConfig_AllNonCompounds,
+  PropTypeConfig_Compound,
+} from '@theatre/core/propTypes'
+import type {PathToProp} from '@theatre/shared/src/utils/addresses'
+import {getPropConfigByPath} from '@theatre/shared/propTypes/utils'
+import {isPlainObject} from 'lodash-es'
+import userReadableTypeOfValue from '@theatre/shared/utils/userReadableTypeOfValue'
+
+function cloneDeepSerializable<T>(v: T): T | undefined {
```

Comment:
Question: Is this from somewhere? Any notes about why it's going to return undefined for an empty object?

Reply:
Added a comment [here](a5cf72cd5781c380cd3a913f6b7577ce7d2a1c07) but I guess the main thing to explain (and perhaps put somewhere in the code) is why `{}` or `undefined` or `null` don't ever get serialized/deserialized in any part of the code that touches the store. I'm making a mental note to do this later.